bson/src/test/unit/org/bson/GenericBsonTest.java

```diff
@@ -261,7 +262,10 @@ private void runParseError() {
             }
         } else if (testDefinitionDescription.startsWith("Top-level") || testDefinitionDescription.startsWith("Binary type")) {
             try {
-                parse(str);
+                BsonDocument document = parse(str);
+                if (isTestOfNullByteInCString(description)) {
```

Comment:
This is kind of gross, but required to get the tests to pass for drivers whose JSON parsers don't include the null byte check: https://github.com/mongodb/specifications/commit/ff5d91fea80b9dcb40fb2fb3533fd1bbcee818b0

Reply:
Would any tests fail if we were to call `encodeToHex(document)` in all of them? If no, then we could remove the condition here and leave it only where the error is checked.

Reply:
All tests passes either way.  I had it this way more to be clear on the intentions. But it seems like it just added to the confusion so I removed it.

Done.